### PR TITLE
Meshes added

### PR DESCRIPTION
**Pull request by devngc**
@mostaphaRoudsari Instead of curves, meshes are provided as outputs for shadows and shaded surfaces. These meshes can be used to create Ecotect like shadow range study using a transparent grey shader in native grasshopper custom preview component.

screenclip

### CODE REVIEW

**Comment by Kristoffernegendahl** [comment]
Great. I have been looking for a feature a couple of times.

**Comment by devngc** [comment]
@mostaphaRoudsari Following is the reason for changing the nickname from Shadow Study to Shadow Range.

I believe shadow studies are one of the most widely used environmental studies the designers perform around the globe. And it's great that we have it in Ladybug. My only concern here is that a  beginner might not intuitively realize that by using multiple sunvectors, one can actually turn this component into a ShadowRange component.

Thanks,
Devang

File: src/Ladybug_Shadow Study.py [hunk 1]
```diff
@@ -117,33 +129,46 @@ def srfPtCalculator(i):
         else:
             for i in range(len(analysisSrfs)):
                 srfPtCalculator(i)
-    
+
```
Comment (mostaphaRoudsari) [inline src/Ladybug_Shadow Study.py]: @devngc do we need these comments to stay here?

**Comment by mostaphaRoudsari** [comment]
@devngc thanks. The main issue with this component is that it will fail if the geometry is complicated. If you're looking to only have shadows in plan you can use Grasshopper's component which does it nicely. Still adding meshes to output is good to have. There are a couple of comments in the code that can be removed. Otherwise it looks good and can be merged in. Changing the name also sounds good to me unless @chriswmackey has an objection.

File: src/Ladybug_Shadow Study.py (hunk 1 above)
Comment (devngc) [inline src/Ladybug_Shadow Study.py]: @mostaphaRoudsari  I made those comments while I was reading the code. Not really necessary.

**Comment by devngc** [comment]
@mostaphaRoudsari Well, I very rarely use plan views for shadow studies. Now that we have those meshes, I can also color them and really highlight them. Something I used to do in Ecotect.

File: src/Ladybug_Shadow Study.py (hunk 1 above)
Comment (mostaphaRoudsari) [inline src/Ladybug_Shadow Study.py]: Then let's remove them and I will merge this.

Comment (devngc) [inline src/Ladybug_Shadow Study.py]: @mostaphaRoudsari I will remove those comments and re-commit.

File: src/Ladybug_Shadow Study.py [hunk 2]
```diff
@@ -426,6 +430,18 @@ def isShadedByFaces(self, surface):
...
-        shade = shaded
-
+        shadowCurves = shadowCrvsCollection
+        shadeCurves = shaded
+        
+        shadow = []
+        shade = []
+        
+        for item in shadowCurves:
+            shadow.append(rs.AddPlanarMesh(item))
```
Comment (mostaphaRoudsari) [inline src/Ladybug_Shadow Study.py]: @devngc sorry that I didn't catch this one. `rhinoscriptsyntax` can be quite slow in cases such as this. You can use Rhinocommon to generate the meshes. Here is a sample code:

```python
import Rhino as rc
meshingPar = rc.Geometry.MeshingParameters.Coarse
mesh = rc.Geometry.Mesh.CreateFromPlanarBoundary(crv, meshingPar)
```

File: src/Ladybug_Shadow Study.py [hunk 3]
```diff
@@ -426,6 +430,18 @@ def isShadedByFaces(self, surface):
...
+            
+        for item in shadeCurves:
+            shade.append(rs.AddPlanarMesh(item))
```
Comment (mostaphaRoudsari) [inline src/Ladybug_Shadow Study.py]: Same here!

Comment (devngc) [inline src/Ladybug_Shadow Study.py]: @mostaphaRoudsari Will revise and revert back

File: src/Ladybug_Shadow Study.py [hunk 4]
```diff
@@ -437,10 +437,14 @@ def isShadedByFaces(self, surface):
         shade = []
         
         for item in shadowCurves:
-            shadow.append(rs.AddPlanarMesh(item))
+            meshingPar = rc.Geometry.MeshingParameters.Coarse
```
Comment (mostaphaRoudsari) [inline src/Ladybug_Shadow Study.py]: @devngc thank you for the update. Why are you creating meshingPar each time? You can just create it once outside the loop and use it to generate the meshes. I can merge it as it is but if you can fix this last issue then that would be the better option?

Comment (devngc) [inline src/Ladybug_Shadow Study.py]: @mostaphaRoudsari Sure. Let me update.

**Comment by mostaphaRoudsari** [comment]
@devngc merged in. Thanks.

For the future you may want to check list comprehension in Python. It can be quite useful for what you have been doing here. Obviously, there is nothing wrong with how you have done it right now.

**Comment by devngc** [comment]
@mostaphaRoudsari Thanks for all the inputs and suggestions. I know about list comprehensions. I have seen you use it at several places. Will definitely try to use it from next time..

**Comment by chriswmackey** [comment]
@devngc ,
This looks great!  I can't wait to try it out soon.
-Chris

**Comment by devngc** [comment]
@chriswmackey Thanks. I take inspiration from you people.
-Devang

**Pull request merged**